protocol/client.go

```diff
@@ -370,21 +391,20 @@ func (client *syncPeerClient) newSyncPeerClient(peerID peer.ID) (proto.V1Client,
 	client.connLock.Lock()
 	defer client.connLock.Unlock()
 
-	var err error
-
-	conn := client.network.GetProtoStream(_syncerV1, peerID)
-	if conn == nil {
-		// create new connection
-		conn, err = client.network.NewProtoConnection(_syncerV1, peerID)
-		if err != nil {
-			client.network.ForgetPeer(peerID, "not support syncer v1 protocol")
+	if conn := client.network.GetProtoStream(_syncerV1, peerID); conn != nil {
+		return proto.NewV1Client(conn), nil
+	}
 
-			return nil, fmt.Errorf("failed to open a stream, err %w", err)
-		}
+	// create new connection
+	conn, err := client.network.NewProtoConnection(_syncerV1, peerID)
+	if err != nil {
+		client.network.ForgetPeer(peerID, "not support syncer v1 protocol")
 
-		// save protocol stream
-		client.network.SaveProtocolStream(_syncerV1, conn, peerID)
```

Comment:
remove `SaveProtocolStream`, create and save from `NewProtoConnection`, only single object, hide detail